<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>wafergen</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>

<link rel="stylesheet" href="styles.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Analyse wafergen qPCR</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="wafergenR.html">wafergen in R</a>
</li>
<li>
  <a href="automation_using_function.html">Automation using function</a>
</li>
<li>
  <a href="visualization.html">Visualization</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">wafergen</h1>

</div>


<div id="wafergen-analysis" class="section level1">
<h1>Wafergen analysis</h1>
<p>This tutorial shows how to analyze wafergen data. Wafergen data is nothing but bigger size of qPCR. So, let’s start with qPCR data analisis.</p>
<div id="qpcr-data-analysis-in-r" class="section level2">
<h2>qPCR data analysis in R</h2>
<p>This section will show analyse qPCR data in R. It is environmental samples in the tutorial. Let’s start with house-keeping gene(16S rRNA).</p>
<div id="load-library" class="section level3">
<h3>Load Library</h3>
<pre class="r"><code>library(reshape2)
library(ggplot2)
library(ggpubr)</code></pre>
<pre><code>## Loading required package: magrittr</code></pre>
</div>
<div id="read-data" class="section level3">
<h3>Read data</h3>
<p>First, let’s read data that we need. It is two runs (chip1 and chip2) with same primer sets. Lanying103018_Chip1.txt and Lanying103018_Chip2.txt are the data files from wafergen run. Also, we need two more files, 1) metadata and 2) information of standard. First, it is user’s responsibility to make metadata. It contains description of samples. meta_data.csv is the example of metadata. In this example, there are two years (2016 and 2017), 24 weeks(0 to 23), four manure treatment (LFM, EFM, SM, Control), and Two tillage(CP and NT but only EFM has this treatment). Second, meta_standard.csv contains which standard was used, which primer should be used and expected count. There are four assay(16S, ermB, ermF, tetM) were used, but let’s start with 16S and ermB.</p>
<pre class="r"><code>#read data 
chip1 &lt;- read.table(&quot;../data/Lanying103018_Chip1.txt&quot;, sep=&quot;\t&quot;, header=T)[,c(&quot;Assay&quot;,&quot;Sample&quot;,&quot;Ct&quot;)]
chip2 &lt;- read.table(&quot;../data/Lanying103018_Chip2.txt&quot;, sep=&quot;\t&quot;, header=T)[,c(&quot;Assay&quot;,&quot;Sample&quot;,&quot;Ct&quot;)]

#get sample info
meta &lt;- read.csv(&quot;../data/meta_data.csv&quot;)

#standard
meta_std &lt;- read.csv(&quot;../data/meta_standard.csv&quot;)</code></pre>
<div id="merge-two-chip-into-one-data-table" class="section level4">
<h4>merge two chip into one data table</h4>
<p>Let’s merge two chip into one data table since we have two run.</p>
<pre class="r"><code>dat &lt;- rbind(chip1,chip2)</code></pre>
</div>
</div>
<div id="set-parameters" class="section level3">
<h3>set parameters</h3>
<p>Let’s start with 16S first. Set target name “16s”. This name sould be matched with the name in meta_standard.csv and data files(Lanying103018_Chip1.txt and Lanying103018_Chip2.txt). Also, we assign column name of sample id that used in the metadata (meta_data.csv) in case different user name it different way. In our case our column name that contains sample name is “sample”.</p>
<pre class="r"><code>name_target = &quot;16s&quot;
sample_id = &quot;sample&quot;</code></pre>
<div id="standard-curve-for-16s" class="section level4">
<h4>standard curve for 16S</h4>
<p>First, we need standard curve. To analyze 16S, we subset the data only 16S primer was used.</p>
<pre class="r"><code>sub &lt;- subset(dat, Assay %in% name_target)</code></pre>
<p>This is information of standard with only 16S rRNA gene</p>
<pre class="r"><code>meta_sub &lt;- subset(meta_std, Assay %in% name_target)</code></pre>
<p>Another sub for only standard.</p>
<pre class="r"><code>data_sub &lt;- subset(sub, Sample %in% meta_sub$Sample)</code></pre>
<p>add expected count for each standard samples</p>
<pre class="r"><code>merged_std &lt;- merge(data_sub, meta_sub, by.x = &quot;Sample&quot;, by.y = &quot;Sample&quot;, all.x=T)</code></pre>
<p>Remove NA</p>
<pre class="r"><code>merged_std_no_NA &lt;- subset(merged_std, Ct != &quot;NA&quot;)</code></pre>
<p>This is Ct values for each standard</p>
<pre class="r"><code>ct &lt;- merged_std_no_NA$Ct </code></pre>
<p>This is log10 based expected count</p>
<pre class="r"><code>count &lt;-  log10(merged_std_no_NA$count+1)</code></pre>
<p>Let’s plot it</p>
<pre class="r"><code>table_for_ggplot &lt;- data.frame(ct = ct, count = count)
ggplot(table_for_ggplot, aes(x= count, y=ct))+geom_point()+geom_smooth(method=&#39;lm&#39;)+labs(x=&quot;Count (log10)&quot;, y = &quot;Ct&quot;)</code></pre>
<p><img src="wafergenR_files/figure-html/unnamed-chunk-12-1.png" width="672" /></p>
<p>It looks good. What is the R square value?</p>
<pre class="r"><code>lm.r &lt;- lm(ct ~ count)

summary(lm.r)$r.squared</code></pre>
<pre><code>## [1] 0.9931072</code></pre>
<p>R square looks good. How about efficiency?</p>
<pre class="r"><code>coef &lt;- coef(lm.r)[2]
eff &lt;- 100*((10^(-1/coef))-1)
(eff)</code></pre>
<pre><code>##    count 
## 83.56041</code></pre>
<p>Efficiency dose not looks great. You can see the data point (count=0) is off. It is common data points with low count (or zero count) are not very good align with. Maybe we can remove some point with low count.</p>
<pre class="r"><code>quality_filtered &lt;- subset(merged_std_no_NA, count &gt; 10)</code></pre>
<p>Let’s calculate R square and efficiency again.</p>
<pre class="r"><code>ct &lt;- quality_filtered$Ct 
count &lt;-  log10(quality_filtered$count+1)
lm.r &lt;- lm(ct ~ count)
summary(lm.r)$r.squared</code></pre>
<pre><code>## [1] 0.9975204</code></pre>
<pre class="r"><code>coef &lt;- coef(lm.r)[2]
eff &lt;- 100*((10^(-1/coef))-1)
(eff)</code></pre>
<pre><code>##    count 
## 87.57528</code></pre>
<p>R square and efficiency looks better. and plot it.</p>
<pre class="r"><code>table_for_ggplot &lt;- data.frame(ct = ct, count = count)
ggplot(table_for_ggplot, aes(x= count, y=ct))+geom_point()+geom_smooth(method=&#39;lm&#39;)+labs(x=&quot;Count (log10)&quot;, y = &quot;Ct&quot;)</code></pre>
<p><img src="wafergenR_files/figure-html/unnamed-chunk-17-1.png" width="672" /></p>
</div>
</div>
<div id="calculate-count-using-standard-curve" class="section level3">
<h3>Calculate count using standard curve</h3>
<p>We are going to use data table only countains with 16S assay with it. Make mean since we run it triplicate.</p>
<pre class="r"><code>casted &lt;- dcast(sub, Sample~Assay, fun=mean, value.var = &quot;Ct&quot;)</code></pre>
<p>To understand which sample is which, add metadata.</p>
<pre class="r"><code>merged_meta &lt;- merge(meta, casted, by.x = sample_id, by.y=&quot;Sample&quot;)</code></pre>
<p>We are going to use linear model regression, but this time inverse x and y axis</p>
<pre class="r"><code>inverse.lm &lt;- lm(count ~ ct)</code></pre>
<p>Get Ct values</p>
<pre class="r"><code>val = merged_meta[,name_target]</code></pre>
<p>Using linear model, calculate count. then, save into column name “temp”</p>
<pre class="r"><code>merged_meta$temp&lt;- 10^predict(inverse.lm ,data.frame(ct = val), interval = &quot;predict&quot;)[,1]</code></pre>
<p>Change column name meaningful</p>
<pre class="r"><code>names(merged_meta)[ncol(merged_meta)] &lt;- paste0(&quot;cal_&quot;, name_target)
names(merged_meta)[ncol(merged_meta)-1] &lt;- paste0(&quot;mean_&quot;, name_target)</code></pre>
<p>We can repeat this process into other target</p>
</div>
</div>
</div>

<p>Copyright &copy; 2019 germslab. All rights reserved.</p>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
